test/integration/master/kms_plugin_mock.go

```diff
@@ -66,6 +68,31 @@ func NewBase64Plugin() (*base64Plugin, error) {
 	return result, nil
 }
 
+func (s *base64Plugin) gracefulStop(giveUpAfter time.Duration) error {
```

Comment:
You are right. After re-reading the doc for GraceFulStop I can see that it is a blocking call.
I had some flakes with this test before, so I assumed that this could have been the place were some race conditions may arise. 
Most likely flakes were due to another sleep that @tallclair pointed out in mustSetupTest, which is now removed.